File: Java/dependency-injection/bastard-injection/src/main/java/com/github/grzesiek_galezowski/DependencyInjectionAfter/_1_ControlFreakRefactoredToDependencyInjection/Inbound/MessageInbound.cs
using DependencyInjectionAfter._1_ControlFreakRefactoredToDependencyInjection.Core;

namespace DependencyInjectionAfter._1_ControlFreakRefactoredToDependencyInjection.Inbound
{
  internal interface IInbound
  {
    void SetDomainLogic(IAcmeProcessingWorkflow processingWorkflow);
    void StartListening();
  }

  class MessageInbound : IInbound
  {
    private IAcmeProcessingWorkflow _processingWorkflow;
    private readonly IInboundSocket _socket;
    private readonly IParsing _parsing;

    public MessageInbound(
      IInboundSocket udpSocket, IParsing binaryParsing)
    {
      _socket = udpSocket;
      _parsing = binaryParsing;
    }

    public void SetDomainLogic(IAcmeProcessingWorkflow processingWorkflow)
    {
      _processingWorkflow = processingWorkflow;
    }
    
    public void StartListening()
    {
      byte[] frameData;
      while (_socket.Receive(out frameData))
      {
        var message = _parsing.ResultFor(frameData);
        if (message != null)
        {
          if (_processingWorkflow != null)
          {
            _processingWorkflow.ApplyTo(message);
          }
        }
      }
    }
  }
}

File: Java/dependency-injection/_03_control-freak-refactored/src/main/java/com/github/grzesiek_galezowski/DependencyInjectionAfter/Inbound/MessageInbound.java
package com.github.grzesiek_galezowski.DependencyInjectionAfter.Inbound;

import com.github.grzesiek_galezowski.DependencyInjectionAfter.Core.ProcessingWorkflow;
import com.github.grzesiek_galezowski.DependencyInjectionAfter.Interfaces.AcmeMessage;

public class MessageInbound implements Inbound {
  private ProcessingWorkflow processingWorkflow;
  private final InboundSocket socket;
  private final Parsing parsing;

  public MessageInbound(
      InboundSocket udpSocket, Parsing binaryParsing) {
    socket = udpSocket;
    parsing = binaryParsing;
  }

  public void setDomainLogic(ProcessingWorkflow processingWorkflow) {
    this.processingWorkflow = processingWorkflow;
  }

  public void startListening() {
    byte[] frameData = new byte[100];
    while (socket.receive(frameData)) {
      AcmeMessage message = parsing.resultFor(frameData);
      if (message != null) {
        if (processingWorkflow != null) {
          processingWorkflow.applyTo(message);
        }
      }
    }
  }
}
